crates/miden-lib/asm/kernels/transaction/lib/epilogue.masm

```diff
@@ -194,6 +206,34 @@ proc.execute_auth_procedure
     dropw dropw dropw dropw
 end
 
+# FEE PROCEDURES
+# =================================================================================================
+
+#! Computes the fee required for the current transaction.
+#!
+#! Inputs:  []
+#! Outputs: [fee_amount]
+proc.compute_fee
+    # get the number of cycles the transaction has taken to execute up this point
+    clk
+    # => [num_current_cycles]
+
+    emit.EPILOGUE_AFTER_COMPUTE_FEE_PROCEDURE
+
+    # estimate the number of cycles the transaction will take
+    add.ESTIMATED_AFTER_COMPUTE_FEE_CYCLES
+    # => [num_tx_cycles]
+
+    ilog2
+    # => [num_estimated_verification_cycles]
```

Comment:
I think this would round the number down right? i.e., 17 -> 5. Ideally, what we'd want is to round the number of cycles to the next power of two and then do the `ilog2`.

Reply:
> I think the simplest way to do this would be by doing `ilog2 add.1`, right?

I think this should work.

> Except when the input is already a power of two.

Correct, though given that we are estimating anyways, I'm not 100% sure we need to handle this case.